src/ProjectTemplates/test/BlazorTemplateTestBase.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.E2ETesting;
+using OpenQA.Selenium;
+using Templates.Test.Helpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Templates.Test
+{
+    public abstract class BlazorTemplateTestBase : BrowserTestBase
+    {
+        public BlazorTemplateTestBase(ProjectFactoryFixture projectFactory, BrowserFixture browserFixture, ITestOutputHelper output)
+            : base(browserFixture, output)
+        {
+            ProjectFactory = projectFactory;
+        }
+
+        public ProjectFactoryFixture ProjectFactory { get; set; }
+
+        protected async Task BuildAndRunTest(Project project)
+        {
+            using var aspNetProcess = project.StartBuiltProjectAsync();
+
+            Assert.False(
+                aspNetProcess.Process.HasExited,
+                ErrorMessages.GetFailedProcessMessageOrEmpty("Run built project", project, aspNetProcess.Process));
+
+            await aspNetProcess.AssertStatusCode("/", HttpStatusCode.OK, "text/html");
+            if (BrowserFixture.IsHostAutomationSupported())
+            {
+                aspNetProcess.VisitInBrowser(Browser);
+                TestBasicNavigation(project);
+            }
+            else
+            {
+                BrowserFixture.EnforceSupportedConfigurations();
+            }
+        }
+
+        protected async Task PublishAndRunTest(Project project)
```

Comment:
Can we keep the structure of the tests in line with the other tests? Adding additional indirection levels makes it hard to troubleshoot issues on failed builds.